presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -748,15 +757,19 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
         if (!filterSchema(tableName.getSchemaName())) {
             return ImmutableList.of();
         }
+
+        Optional<Table> optionalTable;
         try {
-            if (metastore.getTable(new HiveIdentity(session), tableName.getSchemaName(), tableName.getTableName()).isEmpty()) {
-                return ImmutableList.of();
-            }
+            optionalTable = metastore.getTable(new HiveIdentity(session), tableName.getSchemaName(), tableName.getTableName());
         }
         catch (HiveViewNotSupportedException e) {
             // exists, would be returned by listTables from schema
+            return ImmutableList.of(tableName);
         }
-        return ImmutableList.of(tableName);
+        return optionalTable
+                .filter(table -> !hideDeltaLakeTables || !isDeltaLakeTable(table))
```

Comment:
why do we need it here if metastore implementation already does filtering out?

Reply:
Because table listing does not happen in the `information_schema.columns` flow, when schema_name and table_name are predicated to a constant. This is covered by 
```
// list all columns in a table
assertThat(metadata.listTableColumns(session, new SchemaTablePrefix(tableName.getSchemaName(), tableName.getTableName())).keySet())
        .doesNotContain(tableName);
```

assertion in the test code

Reply:
This is going to make listing very slow. We want to do this best effort in the metstore for listing tables. For columns, we can filter out in `listTableColumns` which is already fetching table metadata.

Reply:
@electrum  This does not pull any additional data during listing.
Note that normal listing case is handled directly in the metastore code. Here, we short-circuit to avoid metastore call, so we need to apply filtering.

Reply:
Ah, I see that this is a special case for listing a single table.